WordPress/gutenberg
# Framework: Introduce "Milestone It" GitHub action

Closes #9239

This pull request seeks to introduce a [GitHub Actions](https://github.com/features/actions) Docker image configuration. The intent will be that, combined with [`actions/bin/filter` merged](https://github.com/actions/bin/tree/master/filter#merged), it can be used as an action on the `pull_request` event to assign a merged pull request into the next milestone, creating the milestone if it doesn't already exist.

**Testing Instructions:**

It's a bit difficult. I tested using my own fork of the repository. Then you'll need to create a [personal access token](https://github.com/settings/tokens/new?scopes=public_repo). Then save a [Webhook payload example](https://developer.github.com/v3/activity/events/types/#webhook-payload-example-28) for a PullRequestEvent and assign the `"number"` to a real issue or pull request in the forked repository.

At which point, from the root of Gutenberg you can do:

```
GITHUB_EVENT_PATH=~/Desktop/demo-pr.json GITHUB_TOKEN=... GITHUB_REPOSITORY=aduth/gutenberg sh .github/actions/milestone-it/entrypoint.sh
```

You should expect to see that (a) a milestone for "Gutenberg 5.8" is created and (b) the issue or pull request is assigned to it.
